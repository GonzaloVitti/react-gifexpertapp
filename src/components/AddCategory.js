import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({setCategories}) => {
  const [input, setInput] = useState("");
  const handleInputChange = (e) => {
      setInput(e.target.value);
  };
  const handleSubmit =(e)=>{
      e.preventDefault();
      if(input.trim().length>2){
        setCategories(cats =>[input,...cats]);
        setInput('');
      }
      
  }
  return (
    <form onSubmit={handleSubmit}>
      <input
        id="category"
        type="text"
        value={input}
        onChange={(e) => handleInputChange(e)}
      />

    </form>
  );
};
AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
};
