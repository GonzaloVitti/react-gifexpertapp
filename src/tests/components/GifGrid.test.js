import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");
describe("Pruebas con el GifGrid", () => {
  const category = "CS GO";
  /*let wrapper = shallow(<GifGrid category={category}/>);
  beforeEach(() => {
    //jest.clearAllMocks();
    wrapper = shallow(<GifGrid category={category}/>);
  });*/
  test("Debe funcionar correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    let wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs = [{id:'abc',url:'https://localhost/cualquier/cosa.jpg',title:'Cualquier cosa'}]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    let wrapper = shallow(<GifGrid category={category} />);
    //expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
