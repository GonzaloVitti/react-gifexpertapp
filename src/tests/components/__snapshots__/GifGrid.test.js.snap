// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el GifGrid Debe funcionar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    CS GO
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
